x/auth/client/cli/cli_test.go

```diff
@@ -1024,6 +1024,85 @@ func (s *IntegrationTestSuite) createBankMsg(val *network.Validator, toAddr sdk.
 	return res
 }
 
+func (s *IntegrationTestSuite) TestSignWithMultiSigners() {
+	val0, val1 := s.network.Validators[0], s.network.Validators[1]
+	val0Coin := sdk.NewCoin(fmt.Sprintf("%stoken", val0.Moniker), sdk.NewInt(10))
+	val1Coin := sdk.NewCoin(fmt.Sprintf("%stoken", val1.Moniker), sdk.NewInt(10))
+	_, _, addr1 := testdata.KeyTestPubAddr()
+
+	// Creating a tx with 2 msgs from 2 signers: val0 and val1.
+	txBuilder := val0.ClientCtx.TxConfig.NewTxBuilder()
+	txBuilder.SetMsgs(
+		banktypes.NewMsgSend(val0.Address, addr1, sdk.NewCoins(val0Coin)),
+		banktypes.NewMsgSend(val1.Address, addr1, sdk.NewCoins(val1Coin)),
+	)
+	txBuilder.SetFeeAmount(sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))))
+	txBuilder.SetGasLimit(testdata.NewTestGasLimit())
+	// Set signer_infos for both signers. Note: we use the empty signature hack.
+	txBuilder.SetSignatures(
+		signing.SignatureV2{
+			PubKey:   val0.PubKey,
+			Data:     &signing.SingleSignatureData{SignMode: signing.SignMode_SIGN_MODE_DIRECT},
+			Sequence: 45, // TODO Replace with real seq, this should fail in antehandler
+		},
+		signing.SignatureV2{
+			PubKey:   val1.PubKey,
+			Data:     &signing.SingleSignatureData{SignMode: signing.SignMode_SIGN_MODE_DIRECT},
+			Sequence: 67, // TODO Replace with real seq, this should fail in antehandler
+		},
+	)
+	s.Require().Equal([]sdk.AccAddress{val0.Address, val1.Address}, txBuilder.GetTx().GetSigners())
+
+	// Write the unsigned tx into a file.
+	txJSON, err := val0.ClientCtx.TxConfig.TxJSONEncoder()(txBuilder.GetTx())
+	s.Require().NoError(err)
+	unsignedTxFile := testutil.WriteToNewTempFile(s.T(), string(txJSON))
+
+	// TODO Remove println
+	fmt.Println("== unsigned ==")
+	fmt.Println(string(txJSON))
+
+	// Let val0 sign first the file with the unsignedTx.
+	signedByVal0, err := authtest.TxSignExec(val0.ClientCtx, val0.Address, unsignedTxFile.Name(), "--overwrite", fmt.Sprintf("--%s=direct", flags.FlagSignMode))
+	s.Require().NoError(err)
+	signedByVal0File := testutil.WriteToNewTempFile(s.T(), signedByVal0.String())
+
+	// TODO Remove println
+	fmt.Println("== signed by val0 ==")
+	fmt.Println(signedByVal0.String())
```

Comment:
Since I use `--overwrite`, this overwrites the signer_infos too, which is not good. val0 is signing wrong auth_info_bytes.

```js
{
  "body": {
    "messages": [
      {
        "@type": "/cosmos.bank.v1beta1.MsgSend",
        "from_address": "cosmos1mrfmp648zcmpm0tczjlu639xmelta3jvzdd4v0",
        "to_address": "cosmos16tu8kxnm7elp273pnf855j6k5csg36lvhkn9m0",
        "amount": [{ "denom": "node0token", "amount": "10" }]
      },
      {
        "@type": "/cosmos.bank.v1beta1.MsgSend",
        "from_address": "cosmos1yhaez4gc9vhn2us8m365wj5njtvtye5fp6qnea",
        "to_address": "cosmos16tu8kxnm7elp273pnf855j6k5csg36lvhkn9m0",
        "amount": [{ "denom": "node1token", "amount": "10" }]
      }
    ],
    "memo": "",
    "timeout_height": "0",
    "extension_options": [],
    "non_critical_extension_options": []
  },
  "auth_info": {
    "signer_infos": [
      {
        "public_key": {
          "@type": "/cosmos.crypto.secp256k1.PubKey",
          "key": "A/B+U+aGpPhC25ZCZLwCAjofDokBvxaOZAQIFzqAYp4t"
        },
        "mode_info": { "single": { "mode": "SIGN_MODE_DIRECT" } },
        "sequence": "1"
      }
    ],
    "fee": {
      "amount": [{ "denom": "stake", "amount": "10" }],
      "gas_limit": "100000",
      "payer": "",
      "granter": ""
    }
  },
  "signatures": [
    "+JV5P+uViG2glQA5j0rDlm5nvwLK2V7zZhiHBTZZ8hFdMtEGpg84CFPmMid1CMq5Ea4+eQCSFh1OPCDrswYDGw=="
  ]
}
```